gridcapa-swe-runner-app/src/test/java/com/farao_community/farao/swe/runner/app/dichotomy/shift/SweNetworkShifterTest.java

```diff
@@ -93,9 +128,48 @@ void checkIdccEsPtTargetExchangesCalculatedCorrectly() {
     void checkIdccPtEsTargetExchangesCalculatedCorrectly() {
         Map<String, Double> intialNetPositions = Map.of("10YES-REE------0", 50., "10YFR-RTE------C", 100., "10YPT-REN------W", 150.);
         SweNetworkShifter sweNetworkShifter = new SweNetworkShifter(null, ProcessType.IDCC,
-            DichotomyDirection.PT_ES, null, null, 0., 0., intialNetPositions, null);
+                DichotomyDirection.PT_ES, null, null, 0., 0., intialNetPositions, processConfiguration);
         Map<String, Double> shifts = sweNetworkShifter.getTargetExchanges(1000);
         assertEquals(-1000., shifts.get("ES_PT"), 0.001);
         assertEquals(200., shifts.get("ES_FR"), 0.001);
     }
+
+    @Test
+    void shiftNetworkSuccessTest() throws GlskLimitationException, ShiftingException {
+        Network network = Network.read(networkFileName, getClass().getResourceAsStream(networkFileName));
+        Map<String, Double> intialNetPositions = CountryBalanceComputation.computeSweCountriesBalances(network);
```

Comment:
Do not compute countries balance whith real method while testing shiftNetwork method (to be compliant with unit test principles).
Try to put the expected values directly in the map of intialNetPositions.